addons/hr/tests/test_hr_employee.py

```diff
@@ -161,6 +161,33 @@ def test_employee_member_of_department(self):
             employees.filtered_domain(['!'] + employees._search_part_of_department('in', [True])),
             emp_other + emp_parent)
 
+    def test_batch_user_creation_skips_employees_with_existing_users_email(self):
+        existing_email_employee, new_employee = self.env['hr.employee'].create([
```

Comment:
Does not seem to cover all use cases (would also check for invalid emails, like a "wrong" and False work email values), also please test formatted emails